resources/migrations/001_update_migrations.yaml

```diff
@@ -9383,10 +9383,69 @@ databaseChangeLog:
                   name: notification_id
             indexName: idx_notification_subscription_notification_id
 
+  - changeSet:
+      id: v51.2024-09-05T08:00:03
+      author: qnkhuat
+      comment: Create the channel_template table
+      preConditions:
+        - not:
+          - tableExists:
+              tableName: channel_template
+      changes:
+        - createTable:
+            tableName: channel_template
+            remarks: custom template for the channel
+            columns:
+              - column:
+                  name: id
+                  type: int
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: name
+                  type: varchar(64)
+                  remarks: the name of the template
+                  constraints:
+                    nullable: false
+              - column:
+                  name: channel_type
```

Comment:
I'm kinda bothered by this, since I imagine that template / channel compatibility may be many to one. I haven't followed where the discussion went after this was originally pointed out, and imagine the initial UI won't give much opportunity for reusing templates anyway?

Reply:
yea, I'm not super set on this, but I think it's easier to start with the template being tied to the channel. I'm not fully sure how we make global templates and what they're used for. 

FWIW, I do think we'll have channel hierarchy one day, at that point we can have templates with channel type is :channel/generic which is the root of all channels?

Reply:
Even a hierarchy is a bit limiting, and I think its a bit confusing if a column with this name has values which don't match the channels it joins to.

We could rename the field to `template_type`, and for each channel type we can declare the set of template types that it supports. Renaming a column down the line is probably fine, might be YAGNI.